### PR TITLE
Charlesmchen/webrtc/video

### PR DESCRIPTION
**Pull request by charlesmchen**
PTAL @michaelkirk

### CODE REVIEW

File: Signal/Images.xcassets/mute-selected-wide.imageset/Contents.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
   "images" : [
     {
       "idiom" : "universal",
-      "filename" : "video-active.png",
+      "filename" : "mute-selected-wide.png",
```
Comment (charlesmchen) [inline Signal/Images.xcassets/mute-selected-wide.imageset/Contents.json]: I know you want a different style for the "unselected" icons, but I'm trying to prioritize getting things working first.  I'll return to these icons if we have time.

File: Signal/src/AppDelegate.m [hunk 2]
```diff
@@ -317,7 +317,13 @@ - (void)application:(UIApplication *)application
  */
 - (BOOL)application:(UIApplication *)application continueUserActivity:(nonnull NSUserActivity *)userActivity restorationHandler:(nonnull void (^)(NSArray * _Nullable))restorationHandler
 {
-    DDLogWarn(@"%@ called %s with userActivity: %@, but not yet supported.", self.tag, __PRETTY_FUNCTION__, userActivity);
+    if ([userActivity.activityType isEqualToString:@"INStartVideoCallIntent"]) {
+        [[Environment getCurrent].callService handleCallKitStartVideo];
```
Comment (charlesmchen) [inline Signal/src/AppDelegate.m]: AFAIK this is the only way to observe when user taps video button in CallKit ui.

File: Signal/src/call/CallService.swift [hunk 3]
```diff
@@ -608,7 +610,7 @@ fileprivate let timeoutSeconds = 60
         call.state = .connected
 
         // We don't risk transmitting any media until the remote client has admitted to being connected.
-        peerConnectionClient.setAudioEnabled(enabled: true)
+        peerConnectionClient.setAudioEnabled(enabled: !call.isMuted)
```
Comment (charlesmchen) [inline Signal/src/call/CallService.swift]: Caller could mute the call before callee accepts call, so we should honor current model state.

File: Signal/src/call/CallService.swift [hunk 4]
```diff
@@ -713,7 +715,7 @@ fileprivate let timeoutSeconds = 60
      *
      * Can be used for Incoming and Outgoing calls.
      */
-    func handleToggledMute(isMuted: Bool) {
+    func setIsMuted(isMuted: Bool) {
```
Comment (charlesmchen) [inline Signal/src/call/CallService.swift]: There's a lot of "toggle mute" methods that are actually setters.  I've renamed them appropriately.

File: Signal/src/call/Speakerbox/CallKitCallManager.swift [hunk 5]
```diff
@@ -48,14 +49,22 @@ final class CallKitCallManager: NSObject {
...
         requestTransaction(transaction)
     }
 
+//    func setHasVideo(call: SignalCall, hasVideo: Bool) {
+//        let update = CXCallUpdate()
+//        update.hasVideo = call.hasVideo
+//        
+//        // Report the incoming call to the system
+//        provider.reportCall(with: call.localId, updated update: update)
+//    }
```
Comment (charlesmchen) [inline Signal/src/call/Speakerbox/CallKitCallManager.swift]: Whoops, gotta tidy this up before you review.

File: Signal/src/AppDelegate.m (hunk 2 above)
Comment (michaelkirk) [inline Signal/src/AppDelegate.m]: I'm only a little surprised.

Just to keep in mind the comment below, we'll eventually also use this method to initiate a CallKit voice call when someone chooses to initiate a Signal call from their contacts, or from the "callback" action on the system call screen when a call fails.

File: Signal/src/call/CallService.swift (hunk 3 above)
Comment (michaelkirk) [inline Signal/src/call/CallService.swift]: This is a good change. Did you test that this behaves as expected? E.g. the call is initially muted, but it's possible for the caller to un-mute?

The reason I wanted to double check was that there is some plumbing related to setting up the AudioSession in RTCAudioSession that occurs only once you `setAudioEnabled(enabled: true)`.

File: Signal/src/AppDelegate.m (hunk 2 above)
Comment (charlesmchen) [inline Signal/src/AppDelegate.m]: We should also look into how this fits into having an "intents extension".

File: Signal/src/call/CallService.swift (hunk 3 above)
Comment (charlesmchen) [inline Signal/src/call/CallService.swift]: Yup, this works.

**Comment by charlesmchen** [comment]
> Still TODO of course is the actual display of video. Did you want to merge this independent of that?

Yup.

a) We're committed to shipping video.
b) Small branches!
c) If we ever decided to ship without video, we could hide/disable video with a tiny commit.

File: Signal/src/call/Speakerbox/CallKitCallUIAdaptee.swift [hunk 6]
```diff
@@ -139,7 +153,7 @@ final class CallKitCallUIAdaptee: NSObject, CallUIAdaptee, CXProviderDelegate {
         CallService.signalingQueue.async {
             self.callService.handleOutgoingCall(call).then {
                 action.fulfill()
-            }.catch { error in
+            }.catch { _ in
                 self.callManager.removeCall(call)
```
Comment (charlesmchen) [inline Signal/src/call/Speakerbox/CallKitCallUIAdaptee.swift]: Done.

**Comment by charlesmchen** [comment]
> This is a good change. Did you test that this behaves as expected? E.g. the call is initially muted, but it's possible for the caller to un-mute?

These "pre-mute" changes work for the mute button in the call view, but the callkit UI doesn't seem available on caller's phone while ringing, so the second change is future-proofing against that ever changing.

**Pull request merged**